pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/util/HelixSetupUtils.java

```diff
@@ -64,10 +65,12 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(HelixSetupUtils.class);
 
   public static synchronized HelixManager setup(String helixClusterName, String zkPath,
-      String pinotControllerInstanceId, boolean isUpdateStateModel, boolean enableBatchMessageMode) {
+      String pinotControllerInstanceId, boolean isUpdateStateModel, boolean enableBatchMessageMode,
+      LeadControllerResourceConfig leadControllerResourceConfig) {
 
     try {
-      createHelixClusterIfNeeded(helixClusterName, zkPath, isUpdateStateModel, enableBatchMessageMode);
+      createHelixClusterIfNeeded(helixClusterName, zkPath, isUpdateStateModel, enableBatchMessageMode,
```

Comment:
can u add .a TODO that we should not be creating helix cluster in pinot-only mode? thanks

Reply:
`HelixSetupUtils .setup()` method will only be called by either Helix only mode or dual mode controller (in `setUpHelixController` method of `ControllerStarter`).